### PR TITLE
Add support for ASCIICustom

### LINKED ISSUE #235: Support OSCARCustom
SMASH will introduce customisable outputs, which we should support too. In a first step, I will implement it as another OSCAR file format. In a second step, I will port the binary reader to SPARKX, which will in future also be customisable.
For binary, I will implement the latest format, as well as the future custom format. Older version will not be supported.

### PR DESCRIPTION
**Pull request by NGoetz**
This includes necessary changes to support the SMASH 3.2 ASCIICustom output feature. This closes #235

### CODE REVIEW

File: src/sparkx/Particle.py [hunk 1]
```diff
@@ -246,15 +249,21 @@ def __init__(
...
     def __initialize_from_array(
-        self, input_format: str, particle_array: np.ndarray
+        self, input_format: str, particle_array: np.ndarray, attribute_list: List[str]
     ) -> None:
         """
-        Initialize instance attributes based on the provided input format and array.
+        Initialize instance attributes based on the provided input format and array, and optionally an atttribute list.
 
         Parameters
         ----------
```
Comment (Hendrik1704) [inline src/sparkx/Particle.py:269]: The new `input_format` string is missing in the function documentation.

File: src/sparkx/Particle.py [hunk 2]
```diff
@@ -430,7 +468,10 @@ def __initialize_from_array(
                 # It is important for JETSCAPE particles to compute pdg_valid
                 # here because the charge_from_pdg function depends on
                 # it.
-                self.pdg_valid = PDGID(self.pdg).is_valid
```
Comment (Hendrik1704) [inline src/sparkx/Particle.py:433]: Why did you change this? Is there something wrong with the old version?

**Review by Hendrik1704 — CHANGES_REQUESTED** [review]
Looks good. I only have two small comments/questions.

File: src/sparkx/Particle.py (hunk 2 above)
Comment (NGoetz) [reply src/sparkx/Particle.py:433]: Yes. Imagine we have a custom input which does not provide PDG, then this would fail.

File: src/sparkx/Particle.py (hunk 1 above)
Comment (NGoetz) [reply src/sparkx/Particle.py:269]: Fixed.

File: src/sparkx/Particle.py (hunk 2 above)
Comment (Hendrik1704) [reply src/sparkx/Particle.py:433]: OK

**Review by Hendrik1704 — APPROVED** [review]
Looks good now
